.travis.yml

```diff
@@ -1,15 +1,10 @@
 language: groovy
 
-#The Travis team has approved protobuf 2.4.1 as a "safe" apt package.  2.4.1 is currently the only version available.  See full whitelist here : https://github.com/travis-ci/apt-package-whitelist
-addons:
-  apt:
-    packages:
-      - protobuf-compiler
```

Comment:
2.4 is the latest version available within the approved apt mirrors for travis containers, and 2.5 is the latest version of protoc you can sanely build with apt outside of containers for Ubuntu Precise Pangolan (which we probably wouldn't want to do anyway). A bunch of stdlibc++6 and related library fun; I should know, I tried and had tons of fun 🐼 . 

So..we build from source, and cache, which isn't too bad..